applications/nrf5340_audio/src/utils/device_location.c

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include "device_location.h"
+
+#include <errno.h>
+#include "uicr.h"
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(location, CONFIG_DEVICE_LOCATION_LOG_LEVEL);
+
+static uint32_t static_location;
+
+void device_location_get(enum bt_audio_location *location)
+{
+	*location = (enum bt_audio_location)static_location;
+}
+
+#if CONFIG_DEVICE_LOCATION_SET_RUNTIME
+void device_location_set(enum bt_audio_location location)
+{
+	int ret;
+
+	static_location = location;
+
+	/* Try to write the location to UICR */
+	ret = uicr_location_set(location);
+	if (ret) {
+		LOG_ERR("Unable to write location to UICR: %d", ret);
+	}
+}
```

Comment:
Wouldn't this return -EROFS in the case where we have a UICR value set, then overwrite by holding a vol button?